cdc/puller/sorter/merger.go

```diff
@@ -441,10 +443,39 @@ func runMerger(ctx context.Context, numSorters int, in <-chan *flushTask, out ch
 			}
 
 			if minTemp > minResolvedTs {
-				minResolvedTs = minTemp
-				err := onMinResolvedTsUpdate()
-				if err != nil {
-					return errors.Trace(err)
+				atomic.StoreUint64(&minResolvedTs, minTemp)
+				resolvedTsNotifier.Notify()
+			}
+		}
+	})
+
+	errg.Go(func() error {
+		resolvedTsReceiver, err := resolvedTsNotifier.NewReceiver(time.Second * 1)
+		defer resolvedTsReceiver.Stop()
+
+		if err != nil {
+			if cerrors.ErrOperateOnClosedNotifier.Equal(err) {
+				log.Panic("unexpected error", zap.Error(err))
+			}
+			return errors.Trace(err)
+		}
```

Comment:
check the error first, before the defer function. Besides why `ErrOperateOnClosedNotifier` is an unexpected error?